### ISSUE TITLE
Random Screen Locks

### ISSUE THREAD

**Issue by timmocahill** [mate-desktop/mate-panel#1242]
#### Expected behaviour
Mate is invisible when launching apps or enabling screensaver.

#### Actual behaviour
Mate is preventing apps from launching from Mate components like Marco or from the 'start' menu. Sometimes the screen locks are random.

#### Steps to reproduce the behaviour
`Ctrl-Alt-L` reliably locks the screen, but it still allows a keyboard shortcut to tty1 to be visible. When screensaver comes on, it blanks the screen and `Ctrl-Alt-F1`doesn't display. Other similar events are random. Can reliably prevent the issue from arising by launching terminal from desktop (instead of via Mate panel) and launching all apps from the terminal via command line. `Ctrl-Alt-L` can be recovered by switching to tty1, finding pidof mate-screensaver, and killing it.

#### MATE general version
1.24.0

#### Package version
mate-panel/focal,now 1.24.0-2 amd64
[crash.tar.gz](https://github.com/mate-desktop/mate-panel/files/6896512/crash.tar.gz)

#### Linux Distribution
Linux Mint 5.11.0-25-generic #27~20.04.1-Ubuntu SMP Tue Jul 13 17:41:23 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

#### Link to bugreport of your Distribution (requirement)
https://termbin.com/8zi0

**Comment by timmocahill** [comment]
Had another hang with the same configuration as a youtube video played via a USB headphone (Jabra40). I was able to recover by killing Firefox, in which the video was playing. The video became choppy and garbled and then stopped. The stderr is below:
```
ALSA lib conf.c:5187:(snd_config_expand) Unknown parameters 1
ALSA lib control.c:1379:(snd_ctl_open_noupdate) Invalid CTL sysdefault:1
ALSA lib conf.c:5187:(snd_config_expand) Unknown parameters 2
ALSA lib control.c:1379:(snd_ctl_open_noupdate) Invalid CTL sysdefault:2
ALSA lib pcm_dmix.c:1089:(snd_pcm_dmix_open) unable to open slave
ALSA lib pcm_dmix.c:1089:(snd_pcm_dmix_open) unable to open slave
ALSA lib pcm_dmix.c:1089:(snd_pcm_dmix_open) unable to open slave
ALSA lib pcm_dmix.c:1089:(snd_pcm_dmix_open) unable to open slave
ALSA lib pcm_dmix.c:1089:(snd_pcm_dmix_open) unable to open slave
```
On re-launch of Firefox from terminal window, the following appeared:
`[GFX1-]: More than 1 GPU from same vendor detected via PCI, cannot deduce device`

**Comment by lukefromdc** [comment]
I'm wondering if this is related to random lockups I've had that appear to be GPU hangs (thus not related to MATE) where sound still plays but the keyboard and mouse appear to be unresponsive and nothing on the screen that would be expected to update (e.g a progress dialog) ever changes. Sometimes getting to another v-t used to be able to get out of this, but not recently.

Had this with two different motherboards, two different CPU and RAM setups, but the same 1st-gen AMD GCN card and AMDGPU driver. Your stderr also looks like a GPU hang, in this case followed by a reset. Mine apparently won't reset and hangs for good, forcing a reboot

**Comment by timmocahill** [comment]
While it'd be foolish to rule out hardware, the 2nd event was just another example of a crash. I've had Marco crash me. I've done nothing more than hover over the LM menu and lock up. 

For what it's worth, my hardware is:
Topology: 8-Core model: AMD Ryzen 7 4800H with Radeon Graphics bits: 64 type: MT MCP 
arch: Zen rev: 1 L2 cache: 4096 KiB
Device-1: AMD Navi 10 [Radeon RX 5600 OEM/5600 XT / 5700/5700 XT] 
vendor: Micro-Star MSI driver: amdgpu v: kernel bus ID: 03:00.0 chip ID: 1002:731f 
Device-2: AMD Renoir vendor: Micro-Star MSI driver: amdgpu v: kernel bus ID: 08:00.0 
chip ID: 1002:1636

It came with windows 10 which i went ahead and registered so I can run it again as a VM in the future, but it didn't give me any troubles for the brief time it ran. I also sent my issue to the AMD drivers team.

**Comment by timmocahill** [comment]
@lukefromdc I just noticed something when I plugged in a USB headphone (a small pair of Samsung headphones that came with the cell phone I bought a year ago).
```
ALSA lib conf.c:5187:(snd_config_expand) Unknown parameters 1
ALSA lib control.c:1379:(snd_ctl_open_noupdate) Invalid CTL sysdefault:1
ALSA lib conf.c:5187:(snd_config_expand) Unknown parameters 2
ALSA lib control.c:1379:(snd_ctl_open_noupdate) Invalid CTL sysdefault:2
```
 So same error output. This may help convict software or hardware, but idk which.

**Comment by timmocahill** [comment]
New one from Marco today.

https://termbin.com/xy80

**Comment by timmocahill** [comment]
New one from mate-volume-control today!

https://termbin.com/xsha

**Comment by lukefromdc** [comment]
Note that some of us do not follow links to 3ed party file hosting sites for security reasons. Please post that report here directly

**Comment by timmocahill** [comment]
Fair enough, Luke! I wrongly assumed that the button in Mintreport would have some inherent security involved. I attached tars of the docs containing the marco and volume crashes.

[mate-volume-control.crash.tar.gz](https://github.com/mate-desktop/mate-panel/files/6945276/mate-volume-control.crash.tar.gz)
[marco.crash.tar.gz](https://github.com/mate-desktop/mate-panel/files/6945279/marco.crash.tar.gz)

**Comment by lukefromdc** [comment]
You need to install debug symbols (or install the programs by make install) for these backtraces to be useful. The volume control crash was a sigabrt, but marco got a segfault. These look like two different crashes. With the debug symbols at least the segfault in marco would probably give us a bad line in the source code to look at

**Comment by timmocahill** [comment]
i'm using Linux Mint 20.04 and have tried doing the simple things to get the debug symbols (apt-get install marco-dbg) and remedial web searches are only giving me very broad-based answers (e.g. how to install xorg debug symbols, but nothing specific to Mate software). Do you have a link to instructions and/or the debug symbols that I should install?

**Comment by lukefromdc** [comment]
No I do not, as I have never interacted with Mint's bug reporting system. Basically though in Debian/Ubuntu based distros (mint incluyded) you should be looking for a package named <something>-dbg , maybe install Synaptic package manager (if Mint still offers it) and search for all packages with marco in the name. 

Others here might be able to help.

**Comment by timmocahill** [comment]
Ok, here's what I have for tonight's crash of the Mate-settings-daemon. Of course, I didn't have the debug symbols loaded for it, so feel free to ignore the dump. Just adding for tracking purposes.

[crash.tar.gz](https://github.com/mate-desktop/mate-panel/files/6950066/crash.tar.gz)

**Comment by timmocahill** [comment]
So tonight we crashed when trying to print a simple pdf file. Looks like the notification-area-applet died first. Then mate-panel went a few seconds after that, followed by mate-terminal.... which pretty much killed everything else since I launched all my apps from the terminal.

[mate-terminal.crash.tar.gz](https://github.com/mate-desktop/mate-panel/files/6958113/mate-terminal.crash.tar.gz)
[notification-area-crash.tar.gz](https://github.com/mate-desktop/mate-panel/files/6958099/notification-area-crash.tar.gz)
[mate-panel.crash.tar.gz](https://github.com/mate-desktop/mate-panel/files/6958106/mate-panel.crash.tar.gz)

**Comment by lukefromdc** [comment]
Someone else here will have to work on this, more than I can figure out. It
does look however like you have a problem system. No idea what the underlying
problem is (bad RAM, a corrupted file somewhere, something else?) but I cannot
duplicate this sort of crashiness at all. All I've had are GPU lockups, where you
do not SEE anything crash, the screen just stops responding.

**Comment by timmocahill** [comment]
12 hours ago, I blew up my hard disk and installed Fedora 34. It's running kernel 5.13.8-200, but I have no idea how relevant that is. However, I've run clean all day long doing anything and everything that was crashing me under Mint and all three of the window managers. I traded the ability to use the Type C headphones (they're recognized, but don't work) for the ability operate. I was ready to blame hardware, but it seems more likely to be software.

**Comment by lukefromdc** [comment]
Hard drive itself could have been the issue. If you were getting bad sectors containing files MATE needed to operate, this is exactly what would happen.

**Comment by timmocahill** [comment]
Fair point, Luke. It's an SSD, though, so bad sectors are unlikely to
be the cause. That said, the laydown of the OS itself could've been
corrupted. I made live USB sticks for the install with Etcher and I've
since been unable to use Mint to reformat them (block size mismatches).
When I made the Fedora boot drive, I used the native iso image burner
and haven't had those same problems. It's odd because I've used Etcher
to make RasPi boot disks with no issues. There's always room for
error... :-)
